src/VkLayer_GF_amber_scoop/src/amber_scoop_layer.cc

```diff
@@ -177,6 +178,96 @@ VKAPI_ATTR void VKAPI_CALL vkCmdDrawIndexed(
 
 // Other intercepted vulkan functions.
 
+//
+// Our vkCreateGraphicsPipelines function.
+//
+VKAPI_ATTR VkResult VKAPI_CALL vkCreateGraphicsPipelines(
+    VkDevice device, VkPipelineCache pipelineCache, uint32_t createInfoCount,
+    const VkGraphicsPipelineCreateInfo* pCreateInfos,
+    const VkAllocationCallbacks* pAllocator, VkPipeline* pPipelines) {
+  GlobalData* global_data = GetGlobalData();
+  DeviceData* device_data =
+      global_data->device_map.Get(DeviceKey(device))->get();
+
+  // Call the original function.
+  VkResult result = device_data->vkCreateGraphicsPipelines(
+      device, pipelineCache, createInfoCount, pCreateInfos, pAllocator,
+      pPipelines);
+
+  if (result == VK_SUCCESS) {
```

Comment:
Change to: `if (result != VK_SUCCESS) { return result; }` to reduce indentation.